### PR TITLE
[now dev] Use canary `@now/build-utils` or stable based on builders

### PR DESCRIPTION
**Pull request by styfle**
This should install the correct "channel" of the `@now/build-utils` based on the version of the builder.

If one or more builders uses canary, then the `@now/build-utils@canary` will be used.

### CODE REVIEW

File: test/dev-server.unit.js [hunk 1]
```diff
@@ -104,7 +104,8 @@ test('[DevServer] Does not install builders if there are no builds', async t =>
   process.stdout.addListener('data', handler);
   process.stderr.addListener('data', handler);
 
-  await installBuilders(new Set());
```
Comment (styfle) [inline test/dev-server.unit.js:107]: @TooTallNate There was no yarn (2nd param) being passed to this test so I kept it as undefined.

**Review by leo — APPROVED** [review]
Looks good! But we need a @TooTallNate review.

**Review by TooTallNate — CHANGES_REQUESTED** [review]
Code LGTM. But can we add a test case testing the canary build-utils behavior is working correctly?

**Comment by styfle** [comment]
@TooTallNate I added tests.

I wasn't able to get the tests to pass by reading stdout so I instead opted for exporting the function and testing it directly. This allows me to test many different variations, including a custom builder that doesn't actually exist.

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**